aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -486,6 +556,10 @@ inline void Dispatcher::callBoxed(const OperatorHandle& op, Stack* stack) const
     }
     // keeping the guard alive while executing the kernel
     kernel.callBoxed(op, dispatchKeySet, stack);
+    // track outputs
+    if (C10_UNLIKELY(guard.isActive() && guard.needsOutputs())) {
+      guard.setOutputs(*stack);
```

Comment:
This copies the **whole** stack, not only the outputs for the current operator. Is this intentional? Perf might be better if you only copy the outputs for the current operator.

Reply:
This follows the same approach as `needsInputs()` see `runRecordFunction(guard, op, dispatchKey, *stack);` at line 551. This is to be consistent with existing implementation. If we change the existing interface, there might be other code impacted, so we should get some more feedbacks. Maybe another diff to address this.

Reply:
ok, we can follow up as part of a separate perf investigation and improvement pr

Reply:
using `last(stack, num_outputs)` to get the last `op.schema().returns().size()` items would be better, but ok let's do it in a follow up to address this and inputs too